openmetadata-ui/src/main/resources/ui/src/components/common/Table/Table.tsx

```diff
@@ -146,7 +144,36 @@ const Table = <T extends Record<string, unknown>>(
         },
       });
     }
-  }, [dropdownColumnList, columnDropdownSelections]);
+  }, [
+    dropdownColumnList,
+    columnDropdownSelections,
+    selectedEntityTableColumns,
+    entityKey,
+  ]);
+
+  const tableColumnConfigSelections = useMemo(() => {
+    if (entityKey) {
+      if (selectedEntityTableColumns?.[entityKey]) {
+        return selectedEntityTableColumns[entityKey];
+      } else if (!isFullViewTable) {
+        setPreference({
+          selectedEntityTableColumns: {
+            ...selectedEntityTableColumns,
+            [entityKey]: defaultVisibleColumns ?? [],
+          },
+        });
```

Comment:
What's the reason for us to set it? Like we need to set in case of user changes anything on default right

Reply:
yes, we don't require the setter here, reason while modifying any column. We already have the state data, which we take to perform the action. So at that time the initial localStorage set can happen.